tests/Integration/Behaviour/Features/Context/Domain/CartRule/CartRuleFeatureContext.php

```diff
@@ -0,0 +1,354 @@
+     */
+    public function deleteCartRule(string $cartRuleReference): void
+    {
+        $cartRule = SharedStorage::getStorage()->get($cartRuleReference);
+        $command = new DeleteCartRuleCommand((int) $cartRule->id);
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @When I disable cart rule with reference :cartRuleReference
+     *
+     * @param string $cartRuleReference
+     */
+    public function disableCartRule(string $cartRuleReference): void
+    {
+        $cartRule = SharedStorage::getStorage()->get($cartRuleReference);
+        $command = new ToggleCartRuleStatusCommand((int) $cartRule->id, false);
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @When I enable cart rule with reference :cartRuleReference
+     *
+     * @param string $cartRuleReference
+     */
+    public function enableCartRule(string $cartRuleReference): void
+    {
+        $cartRule = SharedStorage::getStorage()->get($cartRuleReference);
+        $command = new ToggleCartRuleStatusCommand((int) $cartRule->id, true);
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @When I bulk enable cart rules :cartRuleReferences
+     *
+     * @param string $cartRuleReferences
+     *
+     * @throws CartRuleConstraintException
+     */
+    public function bulkEnableCartRules(string $cartRuleReferences): void
+    {
+        $cartRuleIds = [];
+        $cartRuleReferenceArray = explode(',', $cartRuleReferences);
+
+        foreach ($cartRuleReferenceArray as $carRuleReference) {
+            $cartRuleIds[] = SharedStorage::getStorage()->get($carRuleReference)->id;
+        }
+
+        $command = new BulkToggleCartRuleStatusCommand($cartRuleIds, true);
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @When I bulk disable cart rules :cartRuleReferences
+     *
+     * @param string $cartRuleReferences
+     *
+     * @throws CartRuleConstraintException
+     */
+    public function bulkDisableCartRules(string $cartRuleReferences): void
+    {
+        $cartRuleIds = [];
+        $cartRuleReferenceArray = explode(',', $cartRuleReferences);
+
+        foreach ($cartRuleReferenceArray as $carRuleReference) {
+            $cartRuleIds[] = SharedStorage::getStorage()->get($carRuleReference)->id;
+        }
+
+        $command = new BulkToggleCartRuleStatusCommand($cartRuleIds, false);
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @When I bulk delete cart rules :cartRuleReferences
+     *
+     * @param string $cartRuleReferences
+     *
+     * @throws CartRuleConstraintException
+     */
+    public function bulkDeleteCartRules(string $cartRuleReferences): void
+    {
+        $cartRuleIds = [];
+        $cartRuleReferenceArray = explode(',', $cartRuleReferences);
+
+        foreach ($cartRuleReferenceArray as $carRuleReference) {
+            $cartRuleIds[] = SharedStorage::getStorage()->get($carRuleReference)->id;
+        }
```

Comment:
If references are correctly storing IDs it allows using the `referencesToIds` method